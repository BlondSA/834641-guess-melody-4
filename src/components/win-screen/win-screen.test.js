import React from "react";
import renderer from "react-test-renderer";
import WinScreen from "./win-screen.jsx";

describe(`<WinScreen/> render correctly`, () => {
  it(`Render 3 question 0 mistakes`, () => {
    const tree = renderer
      .create(
          <WinScreen
            mistakesCount={0}
            questionCount={3}
            onReplayButtonClick={() => {}}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`Render 3 question 1 mistakes`, () => {
    const tree = renderer
      .create(
          <WinScreen
            mistakesCount={1}
            questionCount={3}
            onReplayButtonClick={() => {}}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`Render 0 question 0 mistakes`, () => {
    const tree = renderer
      .create(
          <WinScreen
            mistakesCount={0}
            questionCount={0}
            onReplayButtonClick={() => {}}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`Render 100 question 5 mistakes`, () => {
    const tree = renderer
      .create(
          <WinScreen
            mistakesCount={0}
            questionCount={100}
            onReplayButtonClick={() => {}}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
