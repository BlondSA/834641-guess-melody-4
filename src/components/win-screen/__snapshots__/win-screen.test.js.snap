// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WinScreen/> render correctly Render 0 question 0 mistakes 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total"
  >
    Вы ответили правильно на 
    0
     вопросов и совершили 
    0
     ошибки
  </p>
  <button
    className="replay"
    onClick={[Function]}
    type="button"
  >
    Сыграть ещё раз
  </button>
</section>
`;

exports[`<WinScreen/> render correctly Render 3 question 0 mistakes 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total"
  >
    Вы ответили правильно на 
    3
     вопросов и совершили 
    0
     ошибки
  </p>
  <button
    className="replay"
    onClick={[Function]}
    type="button"
  >
    Сыграть ещё раз
  </button>
</section>
`;

exports[`<WinScreen/> render correctly Render 3 question 1 mistakes 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total"
  >
    Вы ответили правильно на 
    2
     вопросов и совершили 
    1
     ошибки
  </p>
  <button
    className="replay"
    onClick={[Function]}
    type="button"
  >
    Сыграть ещё раз
  </button>
</section>
`;

exports[`<WinScreen/> render correctly Render 100 question 5 mistakes 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total"
  >
    Вы ответили правильно на 
    100
     вопросов и совершили 
    0
     ошибки
  </p>
  <button
    className="replay"
    onClick={[Function]}
    type="button"
  >
    Сыграть ещё раз
  </button>
</section>
`;
